### PR TITLE
Windows: Make stdin pipes synchronous

### PR DESCRIPTION
**Pull request by ChrisDenton**
Stdin pipes do not need to be used asynchronously within the standard library. This is a first step in making pipes mostly synchronous. 

r? @m-ou-se

### CODE REVIEW

File: library/std/src/sys/windows/process.rs [hunk 1]
```diff
@@ -562,7 +567,12 @@ impl Stdio {
 
 impl From<AnonPipe> for Stdio {
     fn from(pipe: AnonPipe) -> Stdio {
-        Stdio::Pipe(pipe)
+        // Note that it's very important we don't give async handles to child processes.
```
Comment (m-ou-se) [inline library/std/src/sys/windows/process.rs:570]: Might be good to add a link to the relevant github issue.

**Comment by m-ou-se** [comment]
Very nice! r=me

File: library/std/src/sys/windows/process.rs [hunk 2]
```diff
@@ -527,13 +527,18 @@ impl Stdio {
             },
 
             Stdio::MakePipe => {
-                let ours_readable = stdio_id != c::STD_INPUT_HANDLE;
-                let pipes = pipe::anon_pipe(ours_readable, true)?;
+                // If stdin then make synchronous
```
Comment (m-ou-se) [inline library/std/src/sys/windows/process.rs:530]: It could be useful to have a comment here explaining why we need async pipes for stdout and stderr.

**Comment by ChrisDenton** [comment]
Comments updated with a fuller explanation and issue link.

@bors r=m-ou-se

**Pull request merged**